src/ucp/rndv/rndv.c

```diff
@@ -1753,21 +1781,38 @@ static void ucp_rndv_dump(ucp_worker_h worker, uct_am_trace_type_t type,
     const ucp_rndv_rtr_hdr_t *rndv_rtr_hdr = data;
     const ucp_rndv_data_hdr_t *rndv_data   = data;
     const ucp_reply_hdr_t *rep_hdr         = data;
+    ucp_tag_rndv_rts_hdr_t *tag_rts;
+    ucp_am_rndv_rts_hdr_t *am_rts;
+    void *rkey_buf;
 
     switch (id) {
     case UCP_AM_ID_RNDV_RTS:
         ucs_assert(rndv_rts_hdr->sreq.ep_id != UCP_EP_ID_INVALID);
 
-        /* RNDV is supported with TAG protocol only */
-        ucs_assert(rndv_rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG);
+        if (rndv_rts_hdr->flags & UCP_RNDV_RTS_FLAG_AM) {
+            am_rts   = ucs_derived_of(rndv_rts_hdr, ucp_am_rndv_rts_hdr_t);
+            rkey_buf = am_rts + 1;
+
+            snprintf(buffer, max, "AM RNDV_RTS am_id %u ep_id 0x%"PRIx64""
+                    " sreq_id 0x%"PRIx64" address 0x%"PRIx64" size %zu",
+                    am_rts->am.am_id, am_rts->super.sreq.ep_id,
+                    am_rts->super.sreq.req_id, am_rts->super.address,
+                    am_rts->super.size);
 
-        snprintf(buffer, max, "RNDV_RTS tag %"PRIx64" ep_id 0x%"PRIx64
-                 " sreq_id 0x%"PRIx64" address 0x%"PRIx64" size %zu",
-                 rndv_rts_hdr->tag.tag, rndv_rts_hdr->sreq.ep_id,
-                 rndv_rts_hdr->sreq.req_id, rndv_rts_hdr->address,
-                 rndv_rts_hdr->size);
+        } else {
+            ucs_assert(rndv_rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG);
+
+            tag_rts  = ucs_derived_of(rndv_rts_hdr, ucp_tag_rndv_rts_hdr_t);
+            rkey_buf = tag_rts + 1;
+
+            snprintf(buffer, max, "RNDV_RTS tag %"PRIx64" ep_id 0x%"PRIx64""
+                    " sreq_id 0x%"PRIx64" address 0x%"PRIx64" size %zu",
+                    tag_rts->tag.tag, tag_rts->super.sreq.ep_id,
+                    tag_rts->super.sreq.req_id, tag_rts->super.address,
+                    tag_rts->super.size);
+        }
```

Comment:
this could be simplified no need to duplicate the print, just add flags:
"RNDV_RTS %c%c"  , ... , (flags&TAG)?'t':'-', (flags&AM)?'a':'-'

Reply:
i also print either tag or am id, having 2 separate `sprintf`s looks more clear to me

Reply:
better remove duplication (another string buf if needed), if we add more common params to RTS it will have to be added twice